analyzers/src/SonarAnalyzer.CSharp/Rules/CheckArgumentException.cs

```diff
@@ -72,52 +72,47 @@ private static void CheckForIssue(SyntaxNodeAnalysisContext analysisContext)
                 return;
             }
 
-            var parameterNameValue = default(Optional<object>);
-            var messageValue = default(Optional<object>);
-            for (var i = 0; i < methodSymbol.Parameters.Length; i++)
-            {
-                var argument = objectCreation.ArgumentList.Arguments[i];
-                var argumentExpression = objectCreation.ArgumentList.Arguments[i].Expression;
-                RetrieveParameterAndMessageArgumentValue(
-                    argument.NameColon != null
-                        ? argument.NameColon.Name.Identifier.ValueText
-                        : methodSymbol.Parameters[i].MetadataName,
-                    argumentExpression,
-                    analysisContext.SemanticModel,
-                    ref parameterNameValue,
-                    ref messageValue);
-            }
+            var parameterAndMessage = RetrieveParameterAndMessageArgumentValue(methodSymbol, objectCreation, analysisContext.SemanticModel);
 
-            if (!parameterNameValue.HasValue)
+            if (!parameterAndMessage.Item1.HasValue)
             {
                 // can't check non-constant strings OR argument is not set
                 return;
             }
 
             var methodArgumentNames = GetMethodArgumentNames(objectCreation.Expression).ToHashSet();
-            if (!methodArgumentNames.Contains(TakeOnlyBeforeDot(parameterNameValue)))
+            if (!methodArgumentNames.Contains(TakeOnlyBeforeDot(parameterAndMessage.Item1)))
             {
-                var message = messageValue.HasValue && messageValue.Value != null && methodArgumentNames.Contains(TakeOnlyBeforeDot(messageValue))
+                var message = parameterAndMessage.Item2.HasValue && parameterAndMessage.Item2.Value != null && methodArgumentNames.Contains(TakeOnlyBeforeDot(parameterAndMessage.Item2))
```

Comment:
I don't understand the idea of an `Optional` that has a null value...

maybe you can remove the `Optional` boiler plate in this case? it's a private contract anyway